File: C++/PetShop.cpp
#include <iostream> // biar bisa pake input-output kayak cin dan cout
#include <string>   // biar bisa pake tipe data string

using namespace std; // biar nggak perlu ngetik std:: terus

// bikin kelas buat ngatur produk di petshop
class PetShop
{
private:
    string id;       // nyimpen id produk
    string nama;     // nyimpen nama produk
    string kategori; // nyimpen kategori produk
    int harga;       // nyimpen harga produk

public:
    // konstruktor default, kalo nggak ngasih nilai awal
    PetShop()
    {
        this->id = "";       // id dikosongin dulu
        this->nama = "";     // nama juga kosong
        this->kategori = ""; // kategori juga kosong
        this->harga = 0;     // harga diset ke 0
    }

    // konstruktor dengan parameter, biar bisa langsung masukin nilai
    PetShop(string id, string nama, string kategori, int harga)
    {
        this->id = id;             // masukin id yang dikasih
        this->nama = nama;         // masukin nama produk
        this->kategori = kategori; // masukin kategori produk
        this->harga = harga;       // masukin harga produk
    }

    // buat ngambil id produk
    string get_id()
    {
        return this->id; // balikin id-nya
    }

    // buat ganti id produk
    void set_id(string id)
    {
        this->id = id; // update id produk
    }

    // buat ngambil nama produk
    string get_nama()
    {
        return this->nama; // balikin nama produk
    }

    // buat ganti nama produk
    void set_nama(string nama)
    {
        this->nama = nama; // update nama produk
    }

    // buat ngambil kategori produk
    string get_kategori()
    {
        return this->kategori; // balikin kategori produk
    }

    // buat ganti kategori produk
    void set_kategori(string kategori)
    {
        this->kategori = kategori; // update kategori produk
    }

    // buat ngambil harga produk
    int get_harga()
    {
        return this->harga; // balikin harga produk
    }

    // buat ganti harga produk
    void set_harga(int harga)
    {
        this->harga = harga; // update harga produk
    }

    // buat nampilin info produk ke layar
    void display()
    {
        cout << "ID: " << this->id << " | Nama: " << this->nama
             << " | kategori: " << this->kategori << " | harga: " << this->harga << '\n';
        // nampilin id, nama, kategori, sama harga produk
    }

    // destruktor, bakal jalan otomatis kalo objeknya dihapus
    ~PetShop()
    {
    }
};

File: Python/PetShop.py
class PetShop:
    __id = ""  # atribut private buat nyimpen id produk
    __nama = ""  # atribut private buat nyimpen nama produk
    __kategori = ""  # atribut private buat nyimpen kategori produk
    __harga = 0  # atribut private buat nyimpen harga produk

    def __init__(self, id = "", nama = "", kategori = "", harga = 0):  # constructor buat inisialisasi objek
        self.__id = id  # set id produk waktu objek dibuat
        self.__nama = nama  # set nama produk waktu objek dibuat
        self.__kategori = kategori  # set kategori produk waktu objek dibuat
        self.__harga = harga  # set harga produk waktu objek dibuat

    def get_id(self):  # buat ambil id produk
        return self.__id  # balikin nilai id

    def set_id(self, id):  # buat ubah id produk
        self.__id = id  # update id produk

    def get_nama(self):  # buat ambil nama produk
        return self.__nama  # balikin nilai nama

    def set_nama(self, nama):  # buat ubah nama produk
        self.__nama = nama  # update nama produk

    def get_kategori(self):  # buat ambil kategori produk
        return self.__kategori  # balikin nilai kategori

    def set_kategori(self, kategori):  # buat ubah kategori produk
        self.__kategori = kategori  # update kategori produk

    def get_harga(self):  # buat ambil harga produk
        return self.__harga  # balikin nilai harga

    def set_harga(self, harga):  # buat ubah harga produk
        self.__harga = harga  # update harga produk

    def display(self):  # buat nampilin detail produk
        print(f"ID: {self.__id} | Nama: {self.__nama} | Kategori: {self.__kategori} | Harga: {self.__harga}")  # cetak info produk
